raft/server.cc

```diff
@@ -840,14 +840,18 @@ future<> server_impl::io_fiber(index_t last_stable) {
                 co_await _persistence->store_snapshot_descriptor(snp, is_local ? _config.snapshot_trailing : 0);
                 _stats.store_snapshot++;
                 // Drop previous snapshot since it is no longer used
-                 _state_machine->drop_snapshot(old_id);
+                _state_machine->drop_snapshot(old_id);
```

Comment:
If we have separate "drop snapshot" output now there is no reason to treat old snapshot differently. Put `old_id` into `snps_to_drop` and remove it from applied_snapshot.